### PR TITLE
Remove namespace/ dependence on cluster metadata

### PR DESCRIPTION
**Pull request by mmcshane**
**What changed?**
Removes the ClusterInfo interface for a bool parameter to the
cache constructor defining whether or not global namespaces are enabled
on the current cluster. Also refactored CacheEntry functions that relied
on a priori knowledge of the name of the cluster on which they are
currently being executed to taking that information as a parameter from
callers.

**Why?**
Decouples namespace.Cache from cluster metadata and makes it more flexible to use at the same time

**How did you test it?**
Old & new unit tests

**Potential risks**

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/namespace/cache.go [hunk 1]
```diff
@@ -385,15 +373,7 @@ func (c *namespaceCache) refreshLoop() {
...
-func (c *namespaceCache) refreshNamespaces() error {
-	c.refreshLock.Lock()
-	defer c.refreshLock.Unlock()
-	return c.refreshNamespacesLocked()
-}
-
-// this function only refresh the namespaces in the v2 table
-// the namespaces in the v1 table will be refreshed if cache is stale
-func (c *namespaceCache) refreshNamespacesLocked() error {
+func (c *namespaceCache) refreshNamespaces(ctx context.Context) error {
```
Comment (yiminc) [inline common/namespace/cache.go:376]: why is lock no longer needed?

File: common/namespace/cache.go [hunk 2]
```diff
@@ -653,16 +629,15 @@ func (entry *CacheEntry) NotificationVersion() int64 {
...
 
-// IsNamespaceActive return whether the namespace is active, i.e. non global
-// namespace or global namespace which active cluster is the current cluster
-func (entry *CacheEntry) IsNamespaceActive() bool {
+// ActiveInCluster returns whether the namespace is active, i.e. non global
+// namespace or global namespace which active cluster is the provided cluster
+func (entry *CacheEntry) ActiveInCluster(clusterName string) bool {
 	if !entry.isGlobalNamespace {
 		// namespace is not a global namespace, meaning namespace is always
 		// "active" within each cluster
```
Comment (yiminc) [inline common/namespace/cache.go:637]: I thought local namespace is only "active" in its home cluster.

**Review by yiminc — APPROVED** [review]

File: common/namespace/cache.go (hunk 2 above)
Comment (mmcshane) [reply common/namespace/cache.go:637]: Honestly not sure - just keeping the existing logic

File: common/namespace/cache.go (hunk 1 above)
Comment (mmcshane) [reply common/namespace/cache.go:376]: Only ever called from 2 places: (1) in Start() where it completes before the refreshLoop is started, and (2) in the refreshLoop which is a single goroutine. There's no path to concurrent execution.

**Pull request merged**